src/main/java/org/eclipse/tractusx/puris/backend/common/api/controller/RequestApiController.java

```diff
@@ -51,24 +56,32 @@ public abstract class RequestApiController {
     @Autowired
     private RequestService requestService;
 
-    @Autowired
     private RequestApiService requestApiService;
 
     @Autowired
     private ModelMapper modelMapper;
 
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    public RequestApiController(RequestApiService requestApiService) {
+        this.requestApiService = requestApiService;
+    }
+
     /**
      * Receive a request from a consuming partner
      * <p>
-     * Uses the {@link RequestApiService#handleRequest(Request)} method to perform the actual
+     * Uses the {@link RequestApiService#handleRequest(RequestDto)} method to perform the actual
```

Comment:
Is JavaDoc out of sync with this method's signature? 

We should also fix the signature to be the abstract requestDto (preferred as there will be other requests, too) OR remove the abstract request.

Reply:
I removed the abstract class RequestApiController and moved the entire request handling logic to the postRequest method of the ProductStockRequestApiController class.